src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.UserDao;
+import com.dev.cinema.exception.DataProcessingException;
+import com.dev.cinema.lib.anno.Dao;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.List;
+import java.util.Optional;
+import javax.persistence.Query;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long userId = (Long) session.save(user);
```

Comment:
Fetching the id is redundant. We have talked about this on the lesson

Reply:
Thanks, fixed that.